theblues/plans.py

```diff
@@ -54,18 +60,196 @@ def get_plans(self, reference):
                     "%Y-%m-%dT%H:%M:%SZ"
                 ),
                 description=plan.get('description'),
-                price=plan.get('price')), json))
+                price=plan.get('price')), response))
         except (KeyError, TypeError, ValueError) as err:
-            log.info(
+            log.error(
                 'cannot process plans: invalid JSON response: {!r}'.format(
-                    json))
+                    response))
             raise ServerError(
                 'unable to get list of plans for {}: {}'.format(
                     reference.path(), err))
         except Exception as exc:
-            log.info(
+            log.error(
                 'cannot process plans: invalid JSON response: {!r}'.format(
-                    json))
+                    response))
             raise ServerError(
                 'unable to get list of plans for {}: {}'.format(
                     reference.path(), exc))
+
+    def list_wallets(self):
+        """Get the list of wallets
+
+        @return an object containing a list of wallets, a total, and available
+            credit.
+        @raise ServerError
+        """
+        response = make_request(
+            '{}wallet'.format(self.url),
+            timeout=self.timeout,
+            client=self._client)
+        try:
+            response['wallets'] = tuple(map(
+                lambda wallet: Wallet(
+                    owner=wallet['owner'],
+                    wallet=wallet['wallet'],
+                    limit=wallet['limit'],
+                    budgeted=wallet['budgeted'],
+                    unallocated=wallet['unallocated'],
+                    available=wallet['available'],
+                    consumed=wallet['consumed'],
+                    default='default' in wallet), response['wallets']))
+            total = response['total']
+            response['total'] = WalletTotal(
+                limit=total['limit'],
+                budgeted=total['budgeted'],
+                available=total['available'],
+                unallocated=total['unallocated'],
+                usage=total['usage'],
+                consumed=total['consumed'])
+            return response
+        except Exception as err:
+            log.error(
+                'cannot process wallets: invalid JSON response: {!r}'.format(
+                    response))
+            raise ServerError(
+                'unable to get list of wallets: {}'.format(err))
+
+    def get_wallet(self, wallet_name):
+        """Get a single wallet.
+
+        @param the name of the wallet.
+        @return the wallet's total.
+        @raise ServerError
+        """
+        response = make_request(
+            '{}wallet/{}'.format(self.url, wallet_name),
+            timeout=self.timeout,
+            client=self._client)
+        try:
+            total = response['total']
+            response['total'] = WalletTotal(
```

Comment:
Same comment as above avout mutating the response. I think it's better to create a new return value rather than mutating something we don't control.